File: src/main/python/Day1Part2.py
from _io import open

file = open("../resources/inputDay1Part2.txt","r")

numbers = [];

for x in file:
    numbers.append(int(x));

print(numbers)
i = 0
frequency = 0
known = set()

while True:
    frequency += numbers[i]
    i = i+1
    if i >= len(numbers) :
        i = 0
    if frequency in known:
        break
    known.add(frequency)
    
print(frequency)

File: src/main/java/day1/Day1Part2.java
package day1;

import java.io.BufferedReader;
import java.io.IOException;
import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

import utility.FileHelper;

/**
 * @author Kilian
 *
 */
public class Day1Part2 {

	public static void main(String[] args) throws NumberFormatException, IOException {
		BufferedReader br = FileHelper.readFile("inputDay1Part2.txt");
		
		String row = null;
		List<Integer> fileInMemory = new ArrayList<>();
		
		
		//Populate list
		while( (row = br.readLine()) != null) {
			fileInMemory.add(Integer.parseInt(row));
		}
		
		br.close();
		
		System.out.println(findFrequency(fileInMemory));
	}
	
	private static int findFrequency(List<Integer> fileInMemory) {
		Set<Integer> knownFrequencies = new HashSet<Integer>();
		int i = 0;
		int frequency = 0; 
		
		while(true) {
			frequency += fileInMemory.get(i++);
			
			if(!knownFrequencies.add(frequency)) {
				return frequency;
			}
			//Wrap
			if(i >= fileInMemory.size()) {
				i = 0;
			}
		}
	}

}
